mlrun/datastore/sources.py

```diff
@@ -239,6 +233,39 @@ def __init__(
             end_time,
         )
 
+    @property
+    def start_time(self):
+        return self._start_time
+
+    @start_time.setter
+    def start_time(self, start_time):
+        if isinstance(start_time, str):
+            self._start_time = self._convert_to_datetime(start_time)
+        else:
+            self._start_time = start_time
+
+    @property
+    def end_time(self):
+        return self._end_time
+
+    @end_time.setter
+    def end_time(self, end_time):
+        if isinstance(end_time, str):
+            self._end_time = self._convert_to_datetime(end_time)
+        else:
+            self._end_time = end_time
+
+    @staticmethod
+    def _convert_to_datetime(time: str):
+        try:
+            return datetime.fromisoformat(time)
+        except ValueError as error:
+            # check if java's iso format if so remove the Z from the end and try again
+            if time.endswith("Z"):
```

Comment:
I think it makes more sense to first drop a trailing Z if necessary and then parse. Then there is no error handling to consider.

Reply:
Also, shouldn't we transform `Z` to `+00:00` rather than just erase it?